File: Java/0451-0500/455. Assign Cookies.java
// https://leetcode.com/problems/assign-cookies/

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g); Arrays.sort(s);

        int kid = 0, cookie = 0, satisfied = 0;
        
        while(kid < g.length && cookie < s.length){
            if(g[kid] <= s[cookie]){
                satisfied++;kid++;
            }
            cookie++;
        }

        return satisfied;
    }
}

File: Python/0451-0500/455. Assign Cookies.py
# https://leetcode.com/problems/assign-cookies/

class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()

        kid, cookie, satisfied = 0, 0, 0
        
        while kid < len(g) and cookie <len(s):
            if g[kid] <= s[cookie]:
                satisfied += 1
                kid +=1
            
            cookie += 1

        return satisfied
